src/test/java/com/wikia/webdriver/testcases/visualeditor/VEVideoTests.java

```diff
@@ -0,0 +1,170 @@
+        "Short film directed by Guy Ritchie starring David Beckham - H&M Spring 2013";
+
+    String articleName = PageContent.ARTICLE_NAME_PREFIX + DateTime.now().getMillis();
+    VisualEditorPageObject ve =
+        new VisualEditorPageObject(driver).openVEOnArticle(wikiURL, articleName);
+    ve.verifyVEToolBarPresent();
+    ve.verifyEditorSurfacePresent();
+    VisualEditorAddMediaDialog mediaDialog = ve.clickVideoButton();
+    mediaDialog = mediaDialog.searchMedia(mediaTitle);
+    ve = mediaDialog.previewExistingMediaByTitle(mediaTitle);
+    ve.verifyPreviewVideo();
+  }
+
+  @Test(groups = {"VEVideo", "VEMediaResize"})
+  @Execute(asUser = User.USER_9)
+  public void resizeVideoWithHandle() {
+    String articleName = PageContent.ARTICLE_NAME_PREFIX + DateTime.now().getMillis();
+    VisualEditorPageObject ve =
+        new VisualEditorPageObject(driver).openVEOnArticle(wikiURL, articleName);
+    ve.verifyVEToolBarPresent();
+    ve.verifyEditorSurfacePresent();
+    VisualEditorAddMediaDialog mediaDialog = ve.clickVideoButton();
+    mediaDialog = mediaDialog.searchMedia("h");
+    ve = mediaDialog.addExistingMedia(1);
+    ve.verifyVideos(1);
+    Dimension source = ve.getVideoDimension();
+    ve.randomResizeOnMedia();
+    ve.verifyVideoResized(source);
+    VisualEditorSaveChangesDialog save = ve.clickPublishButton();
+    ArticlePageObject article = save.savePage();
+    article.verifyVEPublishComplete();
+  }
+
+  @Test(groups = {"VEVideo", "VEMediaResize"})
+  @Execute(asUser = User.USER_9)
+  public void resizeVideoWithSetting() {
+    int resizeNumber = 250;
+    String articleName = PageContent.ARTICLE_NAME_PREFIX + DateTime.now().getMillis();
+
+    VisualEditorPageObject ve =
+        new VisualEditorPageObject(driver).openVEOnArticle(wikiURL, articleName);
+    ve.verifyVEToolBarPresent();
+    ve.verifyEditorSurfacePresent();
+    VisualEditorAddMediaDialog mediaDialog = ve.clickVideoButton();
+    mediaDialog = mediaDialog.searchMedia("h");
+    ve = mediaDialog.addExistingMedia(1);
+    ve.verifyVideos(1);
+    Dimension source = ve.getVideoDimension();
+    ve.selectMedia();
+    VisualEditorMediaSettingsDialog mediaSettingsDialog = ve.openMediaSettings();
+    mediaSettingsDialog.selectSettings(Setting.ADVANCED);
+    // change width of video to 250
+    mediaSettingsDialog.setCustomSize(resizeNumber, ImageSize.WIDTH);
+    mediaSettingsDialog.clickApplyChangesButton();
+    ve.verifyVideoResized(source);
+    source = ve.getVideoDimension();
+    mediaSettingsDialog = ve.openMediaSettings();
+    mediaSettingsDialog.selectSettings(Setting.ADVANCED);
+    // change height of video to 250
+    mediaSettingsDialog.setCustomSize(resizeNumber, ImageSize.HEIGHT);
+    mediaSettingsDialog.clickApplyChangesButton();
+    ve.verifyVideoResized(source);
+    VisualEditorSaveChangesDialog save = ve.clickPublishButton();
+    ArticlePageObject article = save.savePage();
+    article.verifyVEPublishComplete();
```

Comment:
I would try to separate the code on the meaningful pieces (small methods which will tell me what is going on from the method name).